registration/templates/registration/registration_form.html

```diff
@@ -4,7 +4,7 @@
 {% block title %}{% trans "Register for an account" %}{% endblock %}
 
 {% block content %}
-<form method="post" action="">
+<form method="post" action="{% if request.GET.next %}?next={{request.GET.next}}{% endif %}">
```

Comment:
Since you are dealing with a form, I believe it's cleaner to use a hidden input for `next`

```html
<input type="hidden" name="next" value="{{ request.GET.next }}">
```